src/Symfony/Component/PropertyAccess/Mapping/AttributeMetadataInterface.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyAccess\Mapping;
+
+/**
+ * Stores metadata needed for overriding attributes access methods.
+ *
+ * @internal
+ *
+ * @author Luis Ramón López <lrlopez@gmail.com>
+ */
+interface AttributeMetadataInterface
```

Comment:
Metadata objects are value objects. They should not have an interface. Adding it to the serializer component was a mistake and it's why it is marked as `@internal` (we cannot remove it yet because it will break compatibility).